src/file-events/cpp/win_fsnotifier.cpp

```diff
@@ -42,9 +42,17 @@ void WatchPoint::cancel() {
     if (status == LISTENING) {
         log_fine(server->getThreadEnv(), "Cancelling %s", utf16ToUtf8String(path).c_str());
         status = CANCELLED;
-        BOOL ret = CancelIo(directoryHandle);
+        BOOL ret = CancelIoEx(directoryHandle, &overlapped);
         if (!ret) {
-            throw FileWatcherException("Couldn't cancel I/O", path, GetLastError());
+            status = FINISHED;
+            DWORD lastError = GetLastError();
+            if (lastError == ERROR_NOT_FOUND) {
+                // Do nothing, looks like this is a typical scenario
+                log_fine(server->getThreadEnv(), "Watch point already finished %s", utf16ToUtf8String(path).c_str());
+            } else {
+                // TODO Should we throw here instead?
```

Comment:
We should do the same on Linux and Windows. Is this maybe different when we cancel watching as a request to unregister or when we cancel watching when we close the watcher?

Reply:
I don't understand, sorry. What should we do the same?

If your question is about when do we get `ERROR_NOT_FOUND` – I only saw it only during the stress test, only while terminating the server, and even then only once or twice per execution.

Reply:
On Linux, we currently throw an exception when we can't cancel the watcher. On Windows, we would now only log the problem. I think it would make sense to do the same on both OSes. For example, only logging the problem is fine for me.

Reply:
I see what you mean. I'm about to take a look at Linux now.

Reply:
Let's make this throw on a separate PR.

Reply:
Fixed in https://github.com/gradle/native-platform/pull/142